software-mansion/react-native-screens
# WIP Android keyboard animation test

- TOREVERT: App.js & Test1649 temp changes
- Add stub ScreenModalFragment
- Implement ScreenModalFragment.fragment
- TOREVERT: change stackPresentation in JS example
- Use ScreenModalFragment wheen screen has modal stack presentation set
- Add dependency on material theme 3 and use it in TestExample (potentially TO REVERT)
- linter
- TOREVERT: Bump native deps & kotlin version in TestsExample & lib
- Display modal
- Make StackPresentation.MODAL temporarily transparent to make sure the fragment underneath is not removed
- Allow for dismissal
- Refactor ScreenStack.rootScreen property impl
- Go back button on Second screen in Test1649 (in TE)
- Attempt to dismiss modal via container dismiss method
- TOREVERT(manual): Modify Test1649
- TOREVERT: Log upon receiving native events on JS side
- Bring ScreenModalFragment up to speed with main
- Managed to present the fragment modally
- TOREVERT: temporarily disable custom stack animations
- Kotlinize implementation of `ScreenStack.adapt` method
- Play around with animation
- Stash changes in modal fragment
- Move `recycleView` method to a `ext` module
- Recycle screen before adding it to modal layout
- Another stash of changes :facepalm:
- Set behaviour on screen instead of using dedicated fragment type for standard modal
- Some options for the modal
- Improve view extensions
- Rename dismiss -> dismissFromContainer
- Add more comments into interfaces
- Present 'modal' modally & formSheet as standard bottom sheet
- Stubs, tryGetContext / tryGetActivity, retrieve few props from JS
- MAKE DISMISSAL WORK ON MODAL SHEET
- Throw when fragmentRef.get == null
- Clip bottomsSheet to outline making corner shape visible
- Throw better error message on fragment restoration
- Add some important comments
- Cleanup
- Enable gesture handling in modal view
- Cleanup ModalRootView & nested scroll view compliance
- Add nestedScrollView prop for ScrollView
- Attempt to enforce detent based on props
- Leftover debugging setup
- Comment on STATE_HALF_EXPANDED
- Cleanup example
- Experiment with dimmign view & drag handle
- Stash gesture handling changes
- Various experiments with making dimming view work
- Changes for dimming view animation
- Stash changes
- Stash all changes again
- Add some comment for screen stack
- Whatever
- Fix merge issues
- Naming convention of member fields in ScreensCoordinatorLayout
- Resolve leftover merge conflicts
- Cleanup ScreenStackFragment a bit
- Set corner radius in TE
- Bring back gesture support for standard bottom sheet (see details for bug description)
- Debug & fix logic responsible for dimming view animation
- Allow for navigation to screens above modals
- Improve behaviour when navigating back to screen with modal
- Native support for closing sheet when touched outside
- Improve gesture handling on dimming view
- Update TS types of sheetAllowedDetents & sheetLargestUndimmedDetent props
- sheetInitialDetent -> sheetInitialState
- Update ScreenNativeComponent spec
- Update view manager interfaces for paper
- Bring back functionality of setting sheet height based on fraction passed from JS
- Check against empty array on native side
- Cleanup ScreenStackFragment a bit
- Bring back support for dimming view animation
- Remove unused function
- Update TestExample to use new API
- Add react-versioned files
- Remove RNSModalRootView.kt from main source set
- Revert "Add react-versioned files"
- Revert "Remove RNSModalRootView.kt from main source set"
- Temporarily remove modal functionality
- Fix 3-level detet
- Update TE
- Make ScreensCoordinatorLayout transparent to gestures
- Improve dimming view behaviour a bit
- Small renaming
- Use ViewOutlineProvider to set sheet shape instead of MaterialShapeDrawable
- Do not set cornerradius on MaterialShapeDrawable at all
- Cleanup attachShapeToScreen a bit
- Use GradientDrawable to achieve effect of only top corners being rounded
- Stash changes on rounded corners
- Restore content diplay
- Callbacks for ime inset, set temporarily push presentation as transparent
- Improve sheet animation

## Description



## Changes





## Test code and steps to reproduce



## Checklist
